### ISSUE TITLE
Garnet Aspire Component

### ISSUE THREAD

**Issue by Xen0byte** [dotnet/aspire#3428]
Hi @davidfowl and/or @badrishc, are there any plans that you are aware of to make Garnet work with Aspire?

**Comment by badrishc** [comment]
Garnet can be used with .NET Aspire already:

https://twitter.com/timheuer/status/1773848457247379729

**Comment by davidfowl** [comment]
We can add a first-class resource for garnet in aspire. We don't need a component because stackexchange redis works. @badrishc is there another client we should expose?

**Comment by davidfowl** [comment]
This one is up for grabs if anyone wants to contribute. I recommend looking at the redis resource:

Resource:
https://github.com/dotnet/aspire/tree/main/src/Aspire.Hosting.Redis

Test:
https://github.com/dotnet/aspire/tree/main/tests/Aspire.Hosting.Tests/Redis

Leaving out the component because the redis component exists.

**Comment by Xen0byte** [comment]
> Garnet can be used with .NET Aspire already:
> 
> https://twitter.com/timheuer/status/1773848457247379729

For posterity, this exact snippet didn't work for me. I was running into a bug where the image for download was inheriting the Redis image tag so it couldn't be found. The fix was to explicitly specify the tag.

```csharp
IResourceBuilder<RedisResource> cache = builder.AddRedis("cache", 55513).WithImage("ghcr.io/microsoft/garnet").WithImageTag("latest");
```

**Comment by onionhammer** [comment]
One thing Garnet can support is extensions written in C# - if we don't make a custom component able to load a project which *imports* garnet/runs a server (from a nuget package), writing extensions becomes much harder.

**Comment by Zombach** [comment]
Garnet Work

```csharp
var builder = DistributedApplication.CreateBuilder(args);

const string endpoint = "http";

var api = builder.AddProject<Projects.Api>("api")

var garnet = builder.AddContainer("garnet", "ghcr.io/microsoft/garnet", "latest")
    .WithHttpEndpoint(hostPort: 6379, containerPort: 6379, name: endpoint)
    .GetEndpoint(endpoint);

api.WithReference(garnet);

builder.Build().Run();
```

**Comment by Zombach** [comment]
@davidfowl I would like to try.

**Comment by onionhammer** [comment]
@eerhardt does it support custom extensions or just the pre-built garnet image?

**Comment by eerhardt** [comment]
It just supports an existing garnet container image.

**Comment by Xen0byte** [comment]
Hi. Is there supposed to be an `Aspire.Hosting.Garnet` package up on NuGet? I can see it being mentioned in the README but it doesn't seem to exist. How is the Garnet resource meant to be used? Would someone be able to provide an example?

**Comment by Zombach** [comment]
> Hi. Is there supposed to be an `Aspire.Hosting.Garnet` package up on NuGet? I can see it being mentioned in the resource README, included in the merge commit, but it doesn't seem to exist. How is the Garnet resource meant to be used? Would someone be able to provide an example, please? @Zombach, I feel like I'm missing some key bit of information, but I am not entirely sure what that might be.

[mitchdenny](https://github.com/mitchdenny)
Hello, as @mitchdenny  said, this will all be available in netcore 8.1
[Message](https://github.com/dotnet/aspire/pull/3839#issuecomment-2095812862)

**Comment by mitchdenny** [comment]
Minor correction ... .NET Aspire 8.1 ;)

**Comment by Xen0byte** [comment]
> Hello, as @mitchdenny said, this will all be available in netcore 8.1 [Message](https://github.com/dotnet/aspire/pull/3839#issuecomment-2095812862)

I knew I was missing something important 🙂 many thanks for the clarification.
